File: Local Labor Market Data/Model/Report/CPSReport.swift
//
//  CPSReport.swift
//  Labor Local Data
//
//  Created by Nidhi Chawla on 8/3/18.
//  Copyright © 2018 Department of Labor. All rights reserved.
//

import Foundation


class CPSReport {
    enum lfstCode: String {
        case unemployment = "30"
        case unemploymentRate = "40"
    }
    
    class func getUnemploymentRateSeriesId(forArea area: Area,
                                           adjustment: SeasonalAdjustment) -> SeriesId? {
        guard let nationalArea = area as? National else {return nil}
        
        return getSeriesId(forArea: nationalArea, lfstCode: .unemploymentRate, adjustment: adjustment)
    }
    
    class func getSeriesId(forArea area: National, lfstCode: lfstCode,
                           adjustment: SeasonalAdjustment) -> SeriesId? {
        
        var adjustmentCode = adjustment.rawValue
        adjustmentCode.append(adjustment == .adjusted ? "1" : "0")
        let seriesId = "LN" + adjustmentCode + lfstCode.rawValue + "00000"
        return seriesId
    }

}

File: app/src/main/java/blsapp/dol/gov/blslocaldata/model/reports/CPSReport.kt
package blsapp.dol.gov.blslocaldata.model.reports

import blsapp.dol.gov.blslocaldata.db.entity.AreaEntity


/**
 * CPSReport - UnEmployment Report for National Area
 */

class CPSReport {
    enum class LFSTCode(val code: String) {
        UNEMPLOYMENT("30"),
        UNEMPLOYMENT_RATE("40"),
    }

    companion object {
        fun getUnemploymentRateSeriesId(area: AreaEntity,
                        adjustment: SeasonalAdjustment): SeriesId {
            return getSeriesId(area, lfstCode = LFSTCode.UNEMPLOYMENT_RATE, adjustment = adjustment)
        }

        fun getSeriesId(area: AreaEntity,
                        measureCode: LAUSReport.MeasureCode,
                        adjustment: SeasonalAdjustment): SeriesId {

            val lfstCode: LFSTCode
            if (measureCode == LAUSReport.MeasureCode.UNEMPLOYMENT_RATE) {
                lfstCode = LFSTCode.UNEMPLOYMENT_RATE
            }
            else {
                lfstCode = LFSTCode.UNEMPLOYMENT
            }

            return getSeriesId(area, lfstCode, adjustment)
        }

        fun getSeriesId(area: AreaEntity,
                        lfstCode: LFSTCode,
                        adjustment: SeasonalAdjustment): SeriesId {

            var adjustmentCode = adjustment.value
            adjustmentCode += if (adjustment ==  SeasonalAdjustment.ADJUSTED)  "1" else "0"
            val seriesId = "LN" + adjustmentCode  + lfstCode.code + "00000"
            return seriesId
        }
    }
}